src/bosh-nats-sync/lib/nats_sync/users_sync.rb

```diff
@@ -103,27 +107,21 @@ def query_all_running_vms
       vms
     end
 
-    def call_bosh_api_no_auth(endpoint)
-      response = RestClient::Request.execute(
-        url: @bosh_config['url'] + endpoint,
-        method: :get,
-        verify_ssl: false,
-      )
-      NATSSync.logger.debug(response.inspect)
-      raise("Cannot access: #{endpoint}, Status Code: #{response.code}, #{response.body}") unless response.code == 200
-
-      response.body
-    end
-
     def info
       return @director_info if @director_info
-      body = call_bosh_api_no_auth('/info')
 
-      @director_info = JSON.parse(body)
+      @director_info = JSON.parse(bosh_api_response_body('/info', auth: false))
     end
 
-    def create_authentication_header
-      NATSSync::AuthProvider.new(info, @bosh_config).auth_header
+    def build_headers(auth: true)
+      if auth
+        auth_header = "#{NATSSync::AuthProvider.new(info, @bosh_config).auth_header}"
+        NATSSync.logger.debug 'auth_header is empty, next REST call could fail' if auth_header.empty?
```

Comment:
This used to have a `auth_header.nil?` check as well, which may be necessary if we do not explicitly include something like ActiveSupport to define `#empty?` on `NilClass`.

Reply:
Using `"#{...auth_header}"` should produce `""` in the case `.auth_header` returns `nil`
> `"#{nil}".empty? => true`